packages/scripts/src/addOrbitChain/schemas.ts

```diff
@@ -0,0 +1,357 @@
+    logo: z.string().optional(),
+    description: descriptionSchema,
+  }),
+  nativeTokenData: z
+    .object({
+      name: z.string().min(1),
+      symbol: z.string().min(1),
+      decimals: z.number().int().positive(),
+      logoUrl: z.string().optional(),
+    })
+    .optional(),
+});
+
+export const chainSchema = z
+  .object({
+    chainId: z.number().int().positive(),
+    confirmPeriodBlocks: z.number().int().positive(),
+    ethBridge: ethBridgeSchema,
+    nativeToken: addressSchema.optional(),
+    explorerUrl: urlSchema,
+    rpcUrl: urlSchema,
+    isArbitrum: z.boolean().default(true),
+    isCustom: z.boolean().default(true),
+    isTestnet: z.boolean(),
+    name: z.string().min(1),
+    slug: z.string().min(1),
+    parentChainId: z.number().int().positive(),
+    partnerChainIDs: z.array(z.number().int().positive()).default([]),
+    retryableLifetimeSeconds: z.number().int().positive().default(604800),
+    tokenBridge: tokenBridgeSchema,
+    nitroGenesisBlock: z.number().int().nonnegative().default(0),
+    nitroGenesisL1Block: z.number().int().nonnegative().default(0),
+    depositTimeout: z.number().int().positive().default(1800000),
+    blockTime: z.number().positive().default(0.25),
+    bridgeUiConfig: bridgeUiConfigSchema,
+  })
+  .superRefine(async (chain, ctx) => {
+    const getParentChainInfo = (parentChainId: number) => {
+      switch (parentChainId) {
+        case 1: // Ethereum Mainnet
+          return {
+            rpcUrl: "https://eth.llamarpc.com",
+            blockExplorer: "https://etherscan.io",
+            chainId: 1,
+            name: "Ethereum",
+          };
+        case 42161: // Arbitrum One
+          return {
+            rpcUrl: "https://arb1.arbitrum.io/rpc",
+            blockExplorer: "https://arbiscan.io",
+            chainId: 42161,
+            name: "Arbitrum One",
+          };
+        case 421614: // Arbitrum Sepolia
+          return {
+            rpcUrl: "https://sepolia-rollup.arbitrum.io/rpc",
+            blockExplorer: "https://sepolia.arbiscan.io",
+            chainId: 421614,
+            name: "Arbitrum Sepolia",
+          };
+        default:
+          throw new Error(`Unsupported parent chain ID: ${parentChainId}`);
+      }
+    };
+
+    const parentChainInfo = getParentChainInfo(chain.parentChainId);
+
+    const parentAddressesToCheck = [
+      chain.ethBridge.bridge,
+      chain.ethBridge.inbox,
+      chain.ethBridge.outbox,
+      chain.ethBridge.rollup,
+      chain.ethBridge.sequencerInbox,
+      chain.tokenBridge.parentCustomGateway,
+      chain.tokenBridge.parentErc20Gateway,
+      chain.tokenBridge.parentGatewayRouter,
+      chain.tokenBridge.parentMulticall,
+      chain.tokenBridge.parentProxyAdmin,
+      chain.tokenBridge.parentWeth,
+      chain.tokenBridge.parentWethGateway,
+    ].filter(
+      (address): address is string =>
+        typeof address === "string" &&
+        address !== "0x0000000000000000000000000000000000000000"
```

Comment:
We could make it a global const to make sure we don't make any errors in the future